### PR TITLE
feat(ot3): add HomeRequest, HomeResponse messages

### PR DESCRIPTION
**Pull request by caila-marashaj**

### CODE REVIEW

File: hardware/opentrons_hardware/firmware_bindings/messages/payloads.py [hunk 1]
```diff
@@ -88,6 +88,22 @@ class AddLinearMoveRequestPayload(AddToMoveGroupRequestPayload):
...
+    """Request to home."""
+
+    request_stop_condition: utils.UInt8Field
+    acceleration: utils.Int32Field
+    velocity: utils.Int32Field
+
+
+@dataclass
+class HomeResponse(MoveGroupResponsePayload):
+    """Response to requets to home."""
```
Comment (sfoster1) [inline hardware/opentrons_hardware/firmware_bindings/messages/payloads.py]:
```suggestion
    """Response to requests to home."""
```

**Review by sfoster1 — CHANGES_REQUESTED** [review]
I think this makes sense, although we should make sure we know what the home response thing intends to be. 

A couple things though:

1. When you home, you want to stop _immediately_ when the switch triggers. That's sort of incompatible with acceleration - if you're accelerating, by definition you couldn't immediately stop because you couldn't immediately start
2. when does the response come? after it executes? How is it different from a normal move response?
3. Does this go in move groups? I think it looks like it does, and honestly that's a really good idea that I'm all for

File: hardware/opentrons_hardware/firmware_bindings/messages/payloads.py (hunk 1 above)
Comment (caila-marashaj) [reply hardware/opentrons_hardware/firmware_bindings/messages/payloads.py]: 1- sounds good, I'll take out the acceleration
2- the response is a boolean that comes back after the home move finishes, and just contains a boolean abt whether or not the homing was successful
3- Yes it's going in move groups

File: hardware/opentrons_hardware/firmware_bindings/messages/message_definitions.py [hunk 2]
```diff
@@ -308,6 +308,20 @@ class FirmwareUpdateStatusResponse:  # noqa: D101
...
 
+@dataclass
+class HomeRequest:  # noqa: D101
+    payload: payloads.HomeRequest
+    payload_type: Type[BinarySerializable] = payloads.HomeRequest
+    message_id: Literal[MessageId.home_request] = MessageId.home_request
+
+
+@dataclass
+class HomeResponse:  # noqa: D101
```
Comment (Laura-Danielle) [inline hardware/opentrons_hardware/firmware_bindings/messages/message_definitions.py]: Remove.

File: hardware/opentrons_hardware/firmware_bindings/messages/messages.py [hunk 3]
```diff
@@ -53,6 +53,8 @@
     defs.SetSensorThresholdRequest,
     defs.ReadFromSensorResponse,
     defs.SensorThresholdResponse,
+    defs.HomeRequest,
+    defs.HomeResponse,
```
Comment (Laura-Danielle) [inline hardware/opentrons_hardware/firmware_bindings/messages/messages.py]: Also here.

File: hardware/opentrons_hardware/firmware_bindings/messages/payloads.py [hunk 4]
```diff
@@ -88,6 +88,20 @@ class AddLinearMoveRequestPayload(AddToMoveGroupRequestPayload):
...
 
+@dataclass
+class HomeRequest(AddToMoveGroupRequestPayload):
+    """Request to home."""
+
+    velocity: utils.Int32Field
+
+
+@dataclass
+class HomeResponse(MoveGroupResponsePayload):
```
Comment (Laura-Danielle) [inline hardware/opentrons_hardware/firmware_bindings/messages/payloads.py]: Same here.

**Review by Laura-Danielle — APPROVED** [review]
Looks good to me.

**Review by sfoster1 — APPROVED** [review]
Looks good to me!

**Review by Laura-Danielle — APPROVED** [review]

**Pull request merged**